### DISCUSSION TITLE
How to use Quarto extensions/templates ?

### DISCUSSION THREAD

**Discussion by Andrzej-Andrzej** [quarto-dev/quarto-cli#5394] [Q&A] [answered]
I am slowly making transition from Rmarkdown to Quarto. My questions is how to use quarto extensions like the following:
[https://robjhyndman.com/hyndsight/quarto_templates.html](https://robjhyndman.com/hyndsight/quarto_templates.html)

I installed it correctly, but what do next with it ?

How can I benefit from using those extensions ? Is it for ms word as well or only for pdf ?

Is there any function to list all extensions out that are installed and available ?
Do I need to install a quarto package in R or windows installer of quarto is enough for RStudio ?

How to change font and font size for ms word output ?
When I installed those extension I did not find it in "from template" in RStudio during new qmd file creation. Why is that ?

Thank you in advance.

**Comment by cderv** [comment, marked as answer]
I installed it correctly, but what do next with it ?

You'll find everything about extensions at this part of the documentation 
https://quarto.org/docs/extensions/

This includes details of the type of extensions you can find (filter, shortcode or custom format or revealjs extensions). Depending on what you install, and what the extension is doing, you'll see what to do. 

> Is it for ms word as well or only for pdf ?

It depends on the extension you install and what it does.

> Is there any function to list all extensions out that are installed and available ?

Our website includes a list of known extensions made by us or the community. This repo also list a lot of extensions: https://github.com/mcanouil/awesome-quarto
You can list available installed extension on a project by running `quarto list extensions` - see more at https://quarto.org/docs/extensions/managing.html
There is no repo of extension, so no way to know all that is available on the web - the listing we maintain are already a good source. 

> Do I need to install a quarto package in R or windows installer of quarto is enough for RStudio ?

Windows installer of Quarto is enough to use Quarto. the R package is only useful if you want to call quarto from R. (very specific usage) 

> How to change font and font size for ms word output ?

This is not related to extension. Please look in the documentation about Docx reference doc

> When I installed those extension I did not find it in "from template" in RStudio during new qmd file creation. Why is that ?

Quarto extensions are different from R Markdown templates, and there is no integration in IDE yet. See https://github.com/rstudio/rstudio/issues/11316
With quarto format template can be used from extension using this command : https://quarto.org/docs/extensions/formats.html#format-templates

Hope it helps

**Comment by Andrzej-Andrzej** [comment]
Thank you very much, it helps a lot. As a beginner in quarto I will have to digest everything. 
1. Can you please if possible explain based on this following extension what actually does it do ? Even after reading a description this is not clear to me what does it look like  before and after that extension has been used, please:
[https://quarto-ext.github.io/lightbox/](url)
2. How do I know that a particular extension if for word and the other for html or pdf ? 
3. What is the difference between formats and filters ?

**Comment by dragonstyle** [comment]
- For the Lightbox extension, you can see it in action here:
  <https://quarto-ext.github.io/lightbox/> - click the images in the page and they will zoom to a larger size

- Extensions should include documentation describing how they work and what formats that the support.

- A format packages a variety of files together to target a new or highly customized kind of output (for example a particular journal article). A filter is usually something very general that just modifies a document (for example, you could write a filter to give all links a particular treatment, and this would work for all HTML output without the need for a special format).

**Reply by Andrzej-Andrzej** [reply]
> * For the Lightbox extension, you can see it in action here:
>       https://quarto-ext.github.io/lightbox/ - click the images in the page and they will zoom to a larger size*

and this is it ? Just zooming to a larger size ? My apologies for so simple and basic questions, but I want to understand it.

**Reply by cderv** [reply]
As the README state (https://github.com/quarto-ext/lightbox#readme) 

> An extension that uses the [GLightbox](https://biati-digital.github.io/glightbox/) javascript library to add lightbox styling and behavior to images in your HTML documents.

So yes this is an extension that allows to easily use GLightbox JS Lib in a quarto document. You can look at what the JS lib is doing by following the link: https://biati-digital.github.io/glightbox/

**Reply by mcanouil** [reply]
The best description remains the one from the original JavaScript library which is linked in the extension repository readme.

https://biati-digital.github.io/glightbox/


LightBow Quarto extension repository which describes what it is and how to use it: https://github.com/quarto-ext/lightbox/


Note that so far, about roughly 50 extensions, all of them had a readme describing what the extension is for and how to use/install it.

**Reply by dragonstyle** [reply]
That is the basic idea, though it also displays image descriptions, galleries, and more, as described here. 

<https://github.com/quarto-ext/lightbox>

It is an essentially a modal display of an image or a gallery of images (as the term is traditionally used for websites). It is useful when you'd like to display smaller images in the flow of a document or layout, but allow the user to interact with the images to see larger versions (without requiring the flow of the document to accommodate these full width images).

**Reply by Andrzej-Andrzej** [reply]
Thank you, this behaviour (zooming in) is at present so obvious with pictures placed on webpages that I barely noticed it as a special effect. 
This description below explains it in a simpler way:
"A lightbox displays your images in a front layer while simultaneously darkening the rest of your webpage. It is a popular way to display a larger image without directing your users to another webpage."
[https://howto.caspio.com/tech-tips-and-articles/create-a-lightbox-view-for-images/](url)
